src/goto-programs/remove_function_pointers.cpp

```diff
@@ -408,14 +408,18 @@ void remove_function_pointerst::remove_function_pointer(
       goto_programt::make_goto(call, equal_exprt(pointer, casted_address)));
   }
 
+  goto_programt::targett t;
   // fall-through
   if(add_safety_assertion)
   {
-    goto_programt::targett t =
-      new_code_gotos.add(goto_programt::make_assertion(false_exprt()));
+    t = new_code_gotos.add(goto_programt::make_assertion(false_exprt()));
```

Comment:
FWIW there should probably be an `assume(false)` after this as well.

Reply:
I do agree with this as it  would be consistent with what happens with loop-unwinding assertions (where we add the negated loop condition to the path condition). This would effectively simplify the change here as you'd just unconditionally add the assumption below.